Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -21,49 +28,64 @@ module.exports.Quantity = class Quantity extends Expression
+      @value <= other_v
 
   sameOrAfter: (other) ->
-    if other instanceof Quantity and other.unit == @unit
-      @value >= parseFloat other.value
-    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds >= otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+      @value >= other_v
 
   after: (other) ->
-    if other instanceof Quantity and other.unit == @unit
-      @value > parseFloat other.value
-    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds > otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+      @value > other_v
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit
-      @value < parseFloat other.value
-    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds < otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+      @value < other_v
 
   equals: (other) ->
-    if other instanceof Quantity and @unit == other.unit
-      @value == parseFloat other.value
-    else if other instanceof Quantity and ucum_time_units[other.unit]? and ucum_time_units[@unit]?
-      thisDurationInMilliseconds = durationInMilliseconds(@)
-      otherDurationInMilliseconds = durationInMilliseconds(other)
-      thisDurationInMilliseconds == otherDurationInMilliseconds
-    else null
+    if other instanceof Quantity
+      if (!@unit && other.unit ) || (@unit && !other.unit)
+        false
+      else if !@unit && !other.unit
+        @value == other.value
+      else
+        other_v = ucum.convert(other.value,ucum_unit(other.unit),ucum_unit(@unit))
+        decimalAdjust("round", @value, -8)  == decimalAdjust("round", other_v, -8)
+
+  convertUnits: (to_units) ->
+    convert_value(@value,@unit,to_units)
+
+  dividedBy: (other) ->
+    @multiplyDivied(other,"/")
+
+  multiplyBy: (other) ->
+    @multiplyDivied(other,".") # in ucum . represents multiplication
+
+  multiplyDivied: (other, operator) ->
+    if other instanceof Quantity
+      if @unit and other.unit
+        can_val = @to_ucum()
+        other_can_value = other.to_ucum()
```

Comment:
If you're sure that it will always result in the same base units, regardless of the original units passed in, then it's fine.  My concern is if it parses to different units for similar unit dimensions (e.g., if `kg` parsed down to `g` and `[lb_av]` parsed down to `[oz_av]`, then comparing just the values wouldn't be right).

Perhaps a test or two that compared metric vs non-metric would help confirm one way or the other.  Alternately, we could look at the source code to confirm -- or in the code above do a check to verify the units are the same between `can_val` and `other_can_value`.